platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,333 @@
+   */
+  public ConfigurationUpdater(
+      ConfigurationAdmin ddfConfigAdmin,
+      List strategies,
+      EncryptionService encryptionService,
+      ConfigurationContextFactory configurationContextFactory) {
+    this.ddfConfigAdmin = ddfConfigAdmin;
+    this.strategies = strategies;
+    this.encryptionService = encryptionService;
+    this.pidDataMap = new ConcurrentHashMap<>();
+
+    this.configurationContextFactory = configurationContextFactory;
+  }
+
+  /** @return a read-only map of configuration pids to felix file install properties. */
+  @VisibleForTesting
+  Map<String, CachedConfigData> getPidDataMap() {
+    return Collections.unmodifiableMap(pidDataMap);
+  }
+
+  /**
+   * @see ConfigurationPersistencePlugin
+   * @throws IOException if an error occurs reading configuration data
+   * @throws InvalidSyntaxException technically impossible since {@code null} is being passed as the
+   *     configuration filter. See {@link ConfigurationAdmin#listConfigurations(String)}.
+   */
+  public void init() throws IOException, InvalidSyntaxException {
+    Configuration[] configs = ddfConfigAdmin.listConfigurations(null);
+    if (configs == null) {
+      return;
+    }
+    // TODO: Could we miss configurations?
+    pidDataMap.putAll(
+        Arrays.stream(configs)
+            .map(configurationContextFactory::createContext)
+            .filter(context -> context.getConfigFile() != null)
+            .collect(Collectors.toMap(ConfigurationContext::getServicePid, CachedConfigData::new)));
+  }
+
+  @Override
+  public void handleStore(ConfigurationContext context) throws IOException {
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Object>)
+              () -> {
+                doHandleStore(context);
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
+      // Only caught if the action through a CHECKED exception, thus it MUST be an IOException.
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Sync the config files in etc when an update occurs. There are four cases, which will be denoted
+   * using 2-tuple combinations of variables: [felixFileFromCache, felixFileFromConfig]
+   *
+   * <p>[null, null] (1) Config was created from config admin directly; not by dropping a file.
+   * There is nothing to track.
+   *
+   * <p>[null, X] (2) Some config was just created by dropping a file into etc; track the file.
+   * Update the properties on disk if necessary and perform any special first time processing.
+   *
+   * <p>[Y, null] [Y, X] (3) Felix prop was externally changed or removed. This is an error.
+   *
+   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path. Properties
+   * will be updated on disk if necessary.
+   */
+  private void doHandleStore(ConfigurationContext context) throws IOException {
+    String pid = context.getServicePid();
+    File fileFromConfig = context.getConfigFile();
```

Comment:
✏️  Could we call this `fileFromDisk`? It seems like an odd name, but the way the word `config` is used here is inconsistent with the way it's used on line 214 in the `writeIfNecessary` method.

Reply:
That too would be ambiguous. File from disk makes me think of `etc` but this value is from config admin. I can revise to `fileFromConfigAdmin` if that helps.